src/common/string/filter/sequence-matching.ts

```diff
@@ -0,0 +1,65 @@
+import { fuzzyScore, isPatternInWord } from "./filter";
+
+/**
+ * Determine whether a sequence of letters exists in another string,
+ *   in that order, allowing for skipping. Ex: "chdr" exists in "chandelier")
+ *
+ * filter => sequence of letters
+ * word => Word to check for sequence
+ *
+ * returns either:
+ *    number => if word contains sequence, return a score
+ *    string => if word is empty, return the word (to allow for alphabetical sorting of all available words)
+ *    undefined => if no match was found
+ */
+
+export const fuzzySequentialMatch = (filter: string, ...words: string[]) => {
+  let topScore: string | number | undefined;
```

Comment:
I think we should make this totalscore and combine the scores of all words? And we can just return 0 instead of undefined when no match is found?

Reply:
I am very hesitant to do anything involving injecting our own idea of how scores should work into a scoring system we don't understand.

Also, imagine a case like this:

**Filter:** "bram"

Filter | Word | Score
------------ | ------------ | ------------- |
"bram" | **b**ina**r**y_sensor.b**a**se**m**ent_floor_wet | 6
"bram" | Foo**b**a**r** B**a**se**m**ent Floor Wet | 6
"bram" | Macbook Active | 0
"bram" | binary_sensor.macbook_pro_van_bram_kragten_active | 10

This would mean that the following entities would get the follow total scores:

Filter | Friendly Name | entity_id | Total Score
------------ | ------------ | ------------- | ------------- |
"bram" | Foo**b**a**r** B**a**se**m**ent Floor Wet | **b**ina**r**y_sensor.b**a**se**m**ent_floor_wet | 12
"bram" | Macbook Active | binary_sensor.macbook_pro_van_bram_kragten_active | 10

Reply:
That makes sense